cmd/http/listener_test.go

```diff
@@ -657,11 +657,8 @@ func TestHTTPListenerAcceptError(t *testing.T) {
 				continue
 			}
 
-			s, err := bufio.NewReader(conn).ReadString('\n')
-			if err != nil {
-				t.Errorf("Test %d: reply read: expected = <nil>, got = %s", i+1, err)
-			} else if !strings.Contains(s, "HTTP/1.0 400 Bad Request") {
-				t.Errorf("Test %d: reply read: expected = 'HTTP/1.0 400 Bad Request' got = %s", i+1, s)
+			if _, err = bufio.NewReader(conn).ReadString('\n'); err != io.EOF {
```

Comment:
Unable to understand what problem current check has

Reply:
Well the assumptions changed, the test needs to incorporate that assumption. 

- When there is a routine network timeout we do not write an error to the client which is what this PR fixes. 
- When there is a non-network timeout such as invalid HTTP message or some malformed HTTP we write an error back to the client. 

In this test the client times out @balamurugana so it will see EOF instead of a message from the server.